pkg/controller/signup_test.go

```diff
@@ -276,6 +284,15 @@ func (s *TestSignupSuite) TestVerifyCodeHandler() {
 
 		// Check the status code is what we expect.
 		require.Equal(s.T(), http.StatusOK, rr.Code)
+
+		// Check that the correct userID is passed into the FakeSignupService
+		require.Equal(s.T(), userID, storedUserID)
+
+		// Check that the correct UserSignup is passed into the FakeSignupService
+		require.Equal(s.T(), userID, storedUserSignup.Name)
```

Comment:
We are supposed to update the UserSignup using this FakeSignupService, right? So, just checking the name is not enough. We need to compare the entire UserSignup here. Basically we should make sure that the `Update()` receives the same UserSignup object as `Get()` returned with expected changes (annotations + verificationRequired field in the spec or whatever we update).

Reply:
I've added the checks to confirm all of the set fields, however there is only one UserSignup in play here so of course all the fields will be identical.

Reply:
It doesn't matter how many UserSignups in play here, does it?
If I understand the flow correctly, without that check we could miss a bug like this:
T1: signup := getUserSignup()
T2: VerifyCode(signup) // <- signup is modified here!
T3: signup.Spec.RecauiredVerification = falce // <- bug!!! We should not modify the signup here or use any other signup.
T4: UpdateSignup(signup) // <- we should make sure that the signup here is the same which is returned by VerifyCode() without any further modification.

But if we don't really can test it properly if our mock VerificationCode() doesn't actually modify the signup.

Reply:
I've added a small modification to the UserSignup within the mock verification service, which the test now checks for.